api/src/main/java/org/apache/iceberg/VariantLike.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+/**
+ * Interface for accessing Variant fields.
+ *
+ * <p>This interface supports accessing data in top-level fields, not in nested fields.
+ */
+public interface VariantLike {
+  int size();
+
+  // To access the value of the root element based on the type of provided javaClass.
```

Comment:
need some java docs here and below

Reply:
Added java doc. Can you help checking out the wording?